feat(eventarc): add sample for unmarshalling Cloud Storage event

grayside opened the pull request:
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

muncus commented on eventarc/storage_handler/handler.go:
```diff
@@ -0,0 +1,55 @@
+	ce, err := cloudevents.NewEventFromHTTPRequest(r)
+	if err != nil {
+		log.Printf("failed to parse CloudEvent: %v", err)
+		msg := fmt.Sprintf("%s: expected CloudEvent", http.StatusText(http.StatusBadRequest))
```
(optional): i don't think the `http.StatusText()` is adding much here.
If we use a string literal (even with the same "Bad Request" text), it saves a line in each of these error clauses, making a more concise sample.

grayside replied:
Done.

muncus commented on eventarc/storage_handler/handler_test.go:
```diff
@@ -0,0 +1,132 @@
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// [START eventarc_testing_cloudevent]
```
can we move this region tag up so the imports are included?

grayside replied:
Done.

grayside commented:
In addition to addressing feedback, I added:
* a success test case for an HTTP 200 response code
* removed scoping tricks to reuse a `want` variable, instead switched those to `wantStatus` & `wantBody` variables.
